src/particles/ImpactXParticleContainer.cpp

```diff
@@ -82,4 +82,53 @@ namespace impactx
 //        Redistribute();
 
     }
+
+    void
+    ImpactXParticleContainer::MeanAndStdPositions (
+        amrex::ParticleReal& x_mean, amrex::ParticleReal& x_std,
+        amrex::ParticleReal& y_mean, amrex::ParticleReal& y_std,
+        amrex::ParticleReal& z_mean, amrex::ParticleReal& z_std )
+    {
+        amrex::ParticleReal sum_x, sum_x2, sum_y, sum_y2, sum_z, sum_z2, sum_w;
+
+        using PType = ImpactXParticleContainer::SuperParticleType;
+
+        amrex::ReduceOps<amrex::ReduceOpSum, amrex::ReduceOpSum, amrex::ReduceOpSum, amrex::ReduceOpSum, amrex::ReduceOpSum, amrex::ReduceOpSum, amrex::ReduceOpSum> reduce_ops;
+        auto r = amrex::ParticleReduce<amrex::ReduceData<amrex::ParticleReal, amrex::ParticleReal, amrex::ParticleReal, amrex::ParticleReal, amrex::ParticleReal, amrex::ParticleReal, amrex::ParticleReal>>(
```

Comment:
@ax3l @atmyers Do you know of a way to make these long lines look nicer?
(I could remove the `amrex::` everywhere, but I think that the line would still be quite long.)

Reply:
I think this location is one of the few places where using a macro is justified.

@atmyers does AMReX define an equivalent of `BOOST_PP_REPEAT` already?
https://www.boost.org/doc/libs/1_75_0/libs/preprocessor/doc/ref/repeat.html
https://stackoverflow.com/a/8556064/2719194

Reply:
I think one could use template magic to write a function that, given a typename `T` and int `N`, generates an `amrex::ReduceOps` with `T` repeated `N` times. 

It looks like this PR no longer uses the tuple reduce however? Did you encounter a problem with this @RemiLehe?

Reply:
I think we could still merge together all those reduction operations into one kernel launch using the tuple interface, however. `amrex::ReduceOps` can be a mixture of `ReduceOpMin` and `ReduceOpMax`.

Reply:
Ah, yes, ok. I will update the PR accordingly.

Reply:
> I think one could use template magic to write a function that, given a typename T and int N, generates an amrex::ReduceOps with T repeated N times.

Yes, that;s exactly the `std::make_...` Factory C++ pattern :)